### ISSUE TITLE
Linux (dark system theme): code completion is hard to read

### ISSUE THREAD

**Issue by tmssngr** [eclipse-jdt/eclipse.jdt.ui#410]
I'm on Manjaro Linux (Gnome edition, GTK 3.24.36) which uses a dark theme by default ("Adw-dark"). As shown in the screenshot below, the code completion popup uses colors where foreground (middle gray) and background (middle blue) have a very bad contrast and hence are even for users with reasonable eyes hard to read.
![screenshot](https://user-images.githubusercontent.com/6029561/214033106-4af826fd-733a-4ff3-84fc-01f1bcacfefb.png)
Setting GTK_THEME=Adwaita before launching Eclipse fixes the problem, but it looks like an alien application (white theme while the system is dark).

**Comment by akurtakov** [comment]
Have you tried switching Eclipse theme to Dark too?

**Comment by iloveeclipse** [comment]
@tmssngr: I assume this is not JDT UI issue, and should be reported at https://github.com/eclipse-platform/eclipse.platform.ui.

**Comment by akurtakov** [comment]
Remains to be seen. It could be jdt editor theming too.

**Comment by tmssngr** [comment]
> Have you tried switching Eclipse theme to Dark too?

No, I've used the default. To me as a user it looked like Eclipse already adjusted to the dark system theme.

**Comment by akurtakov** [comment]
Please do so, SWT adjust it's colors accordingly but wherever a plugin overrides/sets some color explicitly one has to switch to Eclipse dark theme.

**Comment by tmssngr** [comment]
In the preferences, page General | Appearance, the checkbox *Enable theming* is already selected as well as *Theme: Dark*. As I have not changed anything in the preferences, this seems like the default on this system.

**Comment by tmssngr** [comment]
BTW, is it expected that all buttons are flat and the enabled and disabled look similar?
![Screenshot from 2023-01-23 14-31-44](https://user-images.githubusercontent.com/6029561/214052086-09caa77b-9b4b-4c28-a059-476aa804f4eb.png)

**Comment by akurtakov** [comment]
![image](https://user-images.githubusercontent.com/574788/214054526-998915cd-a6ec-44c1-946b-d1ebdf79f105.png)

Enabled/disabled are same for me too . The "flat" buttons are some intergration failing with your theme. Note that some years ago "Adwaita theme is guaranteed to work. Eclipse SDK will run with other GTK 3 themes, however we cannot flag these as reference versions without significant community support for testing and/or development of fixes. Bugs that are reproducible only with themes other than Adwaita will be given a lower priority (or may not be fixed at all), compared to bugs which are reproducible on the target environments listed above. " has been put into the project plan https://www.eclipse.org/projects/project-plan.php?planurl=http://www.eclipse.org/eclipse/development/plans/eclipse_project_plan_4_26.xml#target_environments%20plan .
